CONTRIBUTING.md

```diff
@@ -0,0 +1,210 @@
+
+To run them, execute:
+
+```bash
+meson compile cargo-test -C _mesonbuild
+```
+
+Adding tests can be done like in any other rust crate by adding `tests` modules
+and `#[test]` cases directly in code in the same files where the code that should be tested is written.
+
+## Data / Package File Validation
+
+To check the style and correctness of other data/auxiliary files like the `.desktop`
+or `metainfo.xml` AppData definition file execute:
+
+```bash
+meson test -C _mesonbuild
+```
+
+# Contributing Code
+
+All code additions should go through a PR->Review cycle.
+The core maintainers can also push directly to main but should only do that in case of
+trivial changes and fixes.
+
+The CI must run successfully in a opened PR to get it merged.
+Ideally the optional lint step does not report any warnings.
+But because new lints can appear on new clippy versions this is not mandatory.
+
+Please add a short description outlining the changes and the reasons for them.
+When adding new features and/or changes in the UI some screenshots or screen captures would be nice.
+When it fixes a specific issue, the description should reference the to-be-fixed issue with `fixes #<num>`.
+
+# Build system
+
+For building the application the `cargo` calls are wrapped by meson.
+It uses a user specified build directory (e.g. `_mesonbuild`) where the build artifacts will be compiled into.
+All needed additional files needed before/after compilation are prepared and placed into it as well.
+
+# Dependencies
+
+Rust dependencies are declared in the root workspace `Cargo.toml`, or if crate-specific
+in the individual crates `Cargo.toml` configuration files.
+
+The generated `Cargo.lock` file pins the dependencies to specific versions and is checked in.
+
+All non-rust dependencies are declared in the root `meson.build` file.
+You'll see declarations for example for `glib`, `gtk4`, `poppler` and so on.
+
+# Architecture
+
+The codebase is separated into multiple crates that have specific purposes and separate concerns:
+
+- `rnote-compose` : the base crate that is only responsible for supplying basic types needed for a drawing application.
+    Things like shapes, paths, pen-path builders, .. . In it is also the code for how to render these primitives
+    with `cairo` or rather the `piet` abstraction.
+    The dependencies should be kept minimal here. 
+- `rnote-engine` : the core crate of the drawing application.
+    In it is the entire core logic of the drawing part of the Rnote application.
+
+    It is categorized like this:
+    - `rnote-engine/store` : a Entity-Component-System pattern is used there to hold all strokes
+    that are produced by the user in a generational Vector and the methods that define the interactions with them.
+    - `rnote-engine/document` : information about the entire document (it's dimensions, colors, ..)
+    - `rnote-engine/fileformats` : for a stable `.rnote` file format a wrapper for the serialization/deserialization 
```

Comment:
Suggestion: "dictates the current stable Rnote file format, and implements the methods required to load and save itself; additionally contains the code required to convert from and into other formats (notably xopp and older versions of the Rnote file format)"